git-prompt.zsh

```diff
@@ -153,6 +168,29 @@ function _zsh_git_prompt_git_status() {
                 }
                 print RC;
 
+                if (upstream != "" && UPSTREAM_TYPE != "none") {
+                    print UPSTREAM_COLOR;
+                    gsub("origin/", "", upstream);
+                    if (UPSTREAM_TYPE == "auto") {
+                        if (upstream == head) {
```

Comment:
This comparison will fail for branch names with a `%` character, because `%` will be escaped with a double `%%`. You should also escape `%` in the `upstream`. See line 151.